File: shared-libraries/java/localpos-common/src/main/java/com/pds/localpos/common/i18n/ErrorMessageResolver.java
package com.pds.localpos.common.i18n;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.MessageSource;
import org.springframework.stereotype.Component;

import java.util.Locale;

@Slf4j
@Component
@RequiredArgsConstructor
public class ErrorMessageResolver {

    private final MessageSource messageSource;

    public String resolveMessage(String code, Locale locale, Object... args) {
        try {
            return messageSource.getMessage(code, args, locale);
        } catch (Exception e) {
            log.warn("Unable to resolve message for code '{}', locale '{}', args {}: {}", code, locale, args, e.toString());
            return "Message resolve error";
        }
    }
}

File: shared-libraries/kotlin/localpos-common-kt/bin/main/com/pds/localpos/common/i18n/ErrorMessageResolver.kt
package com.pds.localpos.common.i18n

import org.slf4j.LoggerFactory
import org.springframework.context.MessageSource
import org.springframework.stereotype.Component
import java.util.*

@Component
class ErrorMessageResolver(private val messageSource: MessageSource) {

    private val logger = LoggerFactory.getLogger(this::class.java)

    fun resolveMessage(code: String, locale: Locale, vararg args: Any): String {
        return try {
            messageSource.getMessage(code, args, locale)
        } catch (ex: Exception) {
            logger.warn(
                "Unable to resolve message for code '{}', locale '{}', args: {}, error: {}",
                code,
                locale,
                args.contentToString(),
                ex.toString()
            )
            "Message resolve error"
        }
    }
}
